pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -618,16 +618,6 @@ func (d *DeploymentController) setGatewayNameLabel(ti *TemplateInput) {
 		log.Debugf("deployment %s/%s not found in store; using to the new gateway name label", ti.DeploymentName, ti.Namespace)
 		return
 	}
-
-	// Base label choice on the deployment's selector
-	_, exists := dep.(*appsv1.Deployment).Spec.Selector.MatchLabels[constants.DeprecatedGatewayNameLabel]
-	if !exists {
-		// The old label doesn't already exist on the deployment; use the new label
-		return
-	}
-
-	// The old label exists on the deployment; use the old label
-	ti.GatewayNameLabel = constants.DeprecatedGatewayNameLabel
```

Comment:
This is for backwards compatibility from a pretty recent change, not sure we are ready to remove it? @keithmattix

Reply:
Yeah this is still a release or 2 too early; it just got merged in 1.20 IIRC

Reply:
I think I was probably too aggressive; 1 more release should be ok with an upgrade note

Reply:
agree with Keith (at the time of the comment being merged, I had worries about 1.22 being too early). Note Keith added the comment. 

This code costs us nothing but changing now will break users